database/download.go

```diff
@@ -50,25 +60,33 @@ func (db *DB) DownloadByID(ctx context.Context, id primitive.ObjectID) (*Downloa
 
 // DownloadCreate registers a new download. Marks partial downloads by supplying
 // the `bytes` param. If `bytes` is 0 we assume a full download.
-func (db *DB) DownloadCreate(ctx context.Context, user User, skylink Skylink, bytes int64) (*Download, error) {
+func (db *DB) DownloadCreate(ctx context.Context, user User, skylink Skylink, bytes int64) error {
 	if user.ID.IsZero() {
-		return nil, errors.New("invalid user")
+		return errors.New("invalid user")
 	}
 	if skylink.ID.IsZero() {
-		return nil, errors.New("invalid skylink")
+		return errors.New("invalid skylink")
+	}
+
+	// Check if there exists a download of this skylink by this user, updated
+	// within the DownloadUpdateWindow and keep updating that, if so.
+	down, err := db.DownloadRecent(ctx, skylink.ID)
+	if err == nil && down.Updated.Add(DownloadUpdateWindow).After(time.Now().UTC()) {
+		// We found a recent download of this skylink. Let's update it.
+		return db.DownloadIncrement(ctx, down, bytes)
 	}
-	up := Download{
+
+	// We couldn't find a recent download of this skylink, updated within
+	// the DownloadUpdateWindow. We will create a new one.
+	down = &Download{
 		UserID:    user.ID,
 		SkylinkID: skylink.ID,
 		Bytes:     bytes,
-		Timestamp: time.Now().UTC(),
-	}
-	ior, err := db.staticDownloads.InsertOne(ctx, up)
-	if err != nil {
-		return nil, err
+		Created:   time.Now().UTC(),
+		Updated:   time.Now().UTC(),
```

Comment:
Are you updating this field on update? I think but am not 100% sure you are not so `Updated` will always remain the same as `Created`

Reply:
Correct, I wasn't. Fixed.